Src/Common/WebHeaderCollectionExtensions.cs

```diff
@@ -49,7 +49,7 @@ public static void SetNameValueHeaderValue(this NameValueCollection headers, str
                 throw new ArgumentNullException(nameof(keyName));
             }
 
-            headers[headerName] = HeadersUtilities.SetHeaderKeyValue(headers[headerName]?.Split(','), keyName, value);
+            headers[headerName] = string.Join(", ", HeadersUtilities.SetHeaderKeyValue(headers[headerName]?.Split(','), keyName, value));
```

Comment:
Consider revising the name SetHeaderKeyValue. This makes for a weird read where you set and join and set again?
Don't have a good suggestion but may be UpdateHeaderWithKeyValue.